### DISCUSSION TITLE
cross compilation fails for tree-sitter crate

### DISCUSSION THREAD

**Discussion by hgy59** [helix-editor/helix#12175] [Q&A]
There was already an issue with tree-sitter some time ago (reported in #3237).

We use the synology toolchains to build packages for synology devices.

As reported in https://github.com/SynoCommunity/spksrc/pull/6338#issuecomment-2508437852 it is not possible to create the tree-sitter crate except for x64 archs.
The error comes from a CFLAG (-m64) that is valid for x64 archs only.

As there are binaries for aarch64 in the release section (like `helix-24.07-aarch64-linux.tar.xz`) I am just wondering how those are created.

Any instructions are welcome.

**Comment by the-mikedavis** [comment]
The releases are all created by this actions workflow: https://github.com/helix-editor/helix/blob/master/.github/workflows/release.yml

The aarch64 builds use [cross](https://github.com/cross-rs/cross) and target `aarch64-unknown-linux-gnu`
